modules/space/rendering/renderableorbitalkepler.cpp

```diff
@@ -0,0 +1,561 @@
+        "LineWidth",
+        "Line Width",
+        "This value specifies the line width of the trail if the selected rendering "
+        "method includes lines. If the rendering mode is set to Points, this value is "
+        "ignored."
+    };
+    constexpr openspace::properties::Property::PropertyInfo LineColorInfo = {
+        "Color",
+        "Color",
+        "This value determines the RGB main color for the lines and points of the trail."
+    };
+    constexpr openspace::properties::Property::PropertyInfo TrailFadeInfo = {
+        "TrailFade",
+        "Trail Fade",
+        "This value determines how fast the trail fades and is an appearance property. "
+    };
+    static const openspace::properties::Property::PropertyInfo UpperLimitInfo = {
+        "UpperLimit",
+        "Upper Limit",
+        "Upper limit on the number of objects for this renderable, regardless of "
+        "how many objects are contained in the data file"
+    };
+
+    constexpr const char* KeyFile = "Path";
+    constexpr const char* KeyLineNum = "LineNumber";
+}
+
+namespace openspace {
+
+// Count the number of full days since the beginning of 2000 to the beginning of
+// the parameter 'year'
+int RenderableOrbitalKepler::countDays(int year) {
+    // Find the position of the current year in the vector, the difference
+    // between its position and the position of 2000 (for J2000) gives the
+    // number of leap years
+    constexpr const int Epoch = 2000;
+    constexpr const int DaysRegularYear = 365;
+    constexpr const int DaysLeapYear = 366;
+
+    if (year == Epoch) {
+        return 0;
+    }
+
+    // Get the position of the most recent leap year
+    const auto lb = std::lower_bound(LeapYears.begin(), LeapYears.end(), year);
+
+    // Get the position of the epoch
+    const auto y2000 = std::find(LeapYears.begin(), LeapYears.end(), Epoch);
+
+    // The distance between the two iterators gives us the number of leap years
+    const int nLeapYears = static_cast<int>(std::abs(std::distance(y2000, lb)));
+
+    const int nYears = std::abs(year - Epoch);
+    const int nRegularYears = nYears - nLeapYears;
+
+    // Get the total number of days as the sum of leap years + non leap years
+    const int result = nRegularYears * DaysRegularYear + nLeapYears * DaysLeapYear;
+    return result;
+}
+
+// Returns the number of leap seconds that lie between the {year, dayOfYear}
+// time point and { 2000, 1 }
+int RenderableOrbitalKepler::countLeapSeconds(int year, int dayOfYear) {
+    // Find the position of the current year in the vector; its position in
+    // the vector gives the number of leap seconds
+    struct LeapSecond {
+        int year;
+        int dayOfYear;
+        bool operator<(const LeapSecond& rhs) const {
+            return std::tie(year, dayOfYear) < std::tie(rhs.year, rhs.dayOfYear);
+        }
+    };
+
+    const LeapSecond Epoch = { 2000, 1 };
+
+    // List taken from: https://www.ietf.org/timezones/data/leap-seconds.list
```

Comment:
This is making me thing there is probably something interesting we can do with cmake to download the list and generate it? overkill for right now, but a thought for the future

Reply:
Yes, a good idea for a URL sync in the future